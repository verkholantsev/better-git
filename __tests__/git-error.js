// @flow

import GitError from '../src/git-error';

describe('GitError', () => {
    let error;

    beforeEach(() => {
        error = new GitError({ code: 1337, stdout: 'stdout', stderr: 'stderr' });
    });

    it('should contain message property', () => {
        expect(error.message).toMatchSnapshot();
    });

    it('should contain code property', () => {
        expect(error).toHaveProperty('code', 1337);
    });

    it('should contain stdout property', () => {
        expect(error).toHaveProperty('stdout', 'stdout');
    });

    it('should contain stderr property', () => {
        expect(error).toHaveProperty('stderr', 'stderr');
    });
});
