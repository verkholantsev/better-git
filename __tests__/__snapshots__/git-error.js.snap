// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitError should contain message property 1`] = `
"Git error (exit code 1337)
stderr: stderr
stdout: stdout"
`;
